sympy/ntheory/factor_.py

```diff
@@ -2222,30 +2233,90 @@ def is_perfect(n):
     Examples
     ========
 
-    >>> from sympy.ntheory.factor_ import is_perfect, divisors
+    >>> from sympy.ntheory.factor_ import is_perfect, divisors, divisor_sigma
     >>> is_perfect(20)
     False
     >>> is_perfect(6)
     True
-    >>> sum(divisors(6)[:-1])
-    6
+    >>> 6 == divisor_sigma(6) - 6 == sum(divisors(6)[:-1])
+    True
 
     References
     ==========
 
     .. [1] http://mathworld.wolfram.com/PerfectNumber.html
+    .. [2] https://en.wikipedia.org/wiki/Perfect_number
 
     """
     from sympy.core.power import integer_log
 
-    r, b = integer_nthroot(1 + 8*n, 2)
-    if not b:
-        return False
-    n, x = divmod(1 + r, 4)
-    if x:
+    n = as_int(n)
+    if _isperfect(n):
+        return True
+    if n <= PERFECT[MERSENNE_PRIME_EXPONENTS.index(43112609)]:
+        # there may be gaps between this and larger known values
+        # so only conclude in the range for which all values
+        # are known
         return False
-    e, b = integer_log(n, 2)
-    return b and (e + 1) in MERSENNE_PRIME_EXPONENTS
+    if n%2 == 0:
+        last2 = n % 100
+        if last2 != 28 and last2 % 10 != 6:
+            return False
+        r, b = integer_nthroot(1 + 8*n, 2)
+        if not b:
+            return False
+        m, x = divmod(1 + r, 4)
+        if x:
+            return False
+        e, b = integer_log(m, 2)
+        if not b:
+            return False
+    else:
+        if n < 10**1500:
+            return False
+        if n % 105 != 0:
+            return False
+        if not any(n%m == r for m, r in [(12, 1), (468, 117), (324, 81)]):
+            return False
+        facs = factorint(n)
+        odd = [i for i in facs if i%2]
+        if len(odd) != 1:
+            return False
+        # there are properties of q and factors that are not
+        # enumerated here
+        # q = odd[0]
+        del odd
+        if sum(facs.values()) < 101:
+            return False
+        if len(facs) < 10:
+            return False
+        if 3 not in facs and len(facs) < 12:
+            return False
+        c, b, a = sorted(facs)[-3:]
+        if a <= 10**8:
+            return False
+        if b <= 10**4:
+            return False
+        if c <= 100:
+            return False
+        if a > integer_root(3*n, 3):
```

Comment:
does it mean `integer_nthroot` because we dont have any method called `integer_root`.

Reply:
what about this. @asmeurer

Reply:
I rewrote this and committed.